### PR TITLE
Refactor confirmation v2 and add dev buttons

### PR DESCRIPTION
**Pull request by rhasselle-oddball**
## Summary

- Refactor confirmation v2
- Extract all ChapterSectionCollection code away from consumer so it can be reused more easily
- Add dev only buttons for testing

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team-forms#1729

## Testing done

- unit, browser, cypress

## Screenshots

image

clicking "simulate submission"
image

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/simple-forms/shared/components/ConfirmationPageView.v2.jsx [hunk 1]
```diff
@@ -61,30 +76,6 @@ export const ConfirmationPageView = props => {
     [alertRef],
   );
 
-  const form = useSelector(state => state.form);
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/shared/components/ConfirmationPageView.v2.jsx:64]: goal: keep this file as minimal as possible and import things from (eventually forms-system)

File: src/applications/simple-forms/shared/components/ConfirmationPageView.v2.jsx [hunk 2]
```diff
@@ -148,27 +132,11 @@ export const ConfirmationPageView = props => {
...
-            <ChapterSectionCollection
-              chapters={chapters}
-              formData={formData}
-              formConfig={formConfig}
-              pagesFromState={pagesFromState}
-            />
-          </VaAccordionItem>
-        </VaAccordion>
       </div>
+      <ChapterSectionCollection
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/shared/components/ConfirmationPageView.v2.jsx:136]: goal: minimize footprint so easily importable and usable

File: src/applications/simple-forms/shared/components/confirmationPageViewHelpers.jsx [hunk 3]
```diff
@@ -209,26 +223,57 @@ export const buildFields = (chapter, formData, pagesFromState) => {
   });
 };
 
+const useChapterSectionCollection = formConfig => {
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/shared/components/confirmationPageViewHelpers.jsx:226]: logic that was removed from `ConfirmationPageView.v2` is here

File: src/applications/simple-forms/shared/components/confirmationPageViewHelpers.jsx [hunk 4]
```diff
@@ -248,4 +293,80 @@ export const ChapterSectionCollection = ({
...
+          </VaAccordion>
+        </div>
+      </div>
+    );
+  }
+
+  return content;
+};
+
+export const useDevOnlyButtons = ({
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/shared/components/confirmationPageViewHelpers.jsx:315]: decided to add this to help with easy testing on confirmation page. it's optional.

File: src/applications/simple-forms/mock-simple-forms-patterns/containers/ConfirmationPage.jsx [hunk 5]
```diff
@@ -1,99 +1,102 @@
...
+let mockData;
+if (!environment.isProduction()) {
+  mockData = require('../tests/e2e/fixtures/data/default.json');
+}
+
 const USE_CONFIRMATION_PAGE_V2 = true;
 
-export class ConfirmationPage extends React.Component {
-  componentDidMount() {
+const ConfirmationPage = ({ route }) => {
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/mock-simple-forms-patterns/containers/ConfirmationPage.jsx:18]: change to functional component. most code in this file is similar

**Review by rhasselle-oddball — COMMENTED** [review]
dev notes

**Review by ndsprinkle — APPROVED** [review]

**Pull request merged**